test/cri-containerd/main.go

```diff
@@ -195,3 +198,59 @@ func findTestNvidiaGPUDevice() (string, error) {
 	}
 	return strings.TrimSpace(string(out)), nil
 }
+
+// Checks to see if the user has the New-CredentialSpec cmdlet installed.
+func hasCmdlet() error {
+	cmdLet := "New-CredentialSpec"
+	input := fmt.Sprintf("Get-Command %s", cmdLet)
+	_, err := exec.Command(
+		"powershell",
+		input,
+	).Output()
+	if err != nil {
+		return fmt.Errorf("necessary cmdlet not installed. Please run 'Install-Module %s'", cmdLet)
+	}
+	return nil
+}
+
+// Generates a gmsa credential spec and places it at 'path'. Dir has to exist it will not be created by
+// New-CredentialSpec. Could extend to os.Stat and if it doesn't exist could mkdir but will just use ContainerPlat
+// dir for now.
+func generateCredSpec(path string) error {
+	cmd := fmt.Sprintf(`New-CredentialSpec -AccountName %s -Path "%s"`, testGmsaAccount, path)
+	_, err := exec.Command(
+		"powershell",
+		cmd,
+	).Output()
+	if err != nil {
+		return fmt.Errorf("the account specified does not exist")
+	}
+	return nil
+}
+
+func readCredSpec(path string) (string, error) {
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+// Runs all checks necessary to see if user is able to run the gmsa tests. Returns the cred
+// spec and an error if any. Don't need to check if user has access to the account as a failure
+// at generating the cred spec is a sure fire method.
+func gmsaSetup(path string) (string, error) {
+	if err := hasCmdlet(); err != nil {
```

Comment:
I'd rather you not have this func and just return whatever error you get from generateCredSpec.

Reply:
This is more a convenience function to let the user known they need the cmdlet to run these tests, instead of execing the command and parsing the output to see if the error is one of 
1. User doesnt have the cmdlet 
2. User does have the cmdlet but no account exists with this name in the domain

Reply:
Thought splitting the logic would be nicer

Reply:
Why do you need to parse the error? Why not just return any error?

Reply:
For example, the tool could be broken or a parameter deprecated in the future that would return an error even when an account exists. With this design you'd only ever catch an error from whether or not the tool is installed.

Reply:
Good point. I'll skip checking if it exists and just call directly